spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryConfiguration.java

```diff
@@ -117,7 +117,11 @@ public BinderTypeRegistry binderTypeRegistry(ConfigurableApplicationContext conf
 				URL url = resources.nextElement();
 				UrlResource resource = new UrlResource(url);
 				for (BinderType binderType : parseBinderConfigurations(classLoader, resource)) {
-					binderTypes.put(binderType.getDefaultName(), binderType);
+					// Exclude TestSupportBinder when detected as the TestSupport binder is expected to use its own
+					// BinderFactory when used.
+					if (!binderType.getDefaultName().equalsIgnoreCase("test")) {
```

Comment:
right, but the `BinderTypeRegistry` would always get all the binders from `classpath` even if `TestSupportBinderAutoConfiguration` is disabled and the `DefaultBinderFactory` is used.